tests/BridgeImplementationTest.php

```diff
@@ -1,188 +1,204 @@
+		$paramsSeen = array();
+
+		$allowedTypes = array(
+			'text',
+			'number',
+			'list',
+			'checkbox'
+		);
+
+		foreach($this->obj::PARAMETERS as $context => $params) {
+			if ($multiContexts) {
+				$this->assertInternalType('string', $context, 'invalid context name');
+				$this->assertNotEmpty($context, 'empty context name');
+			}
 
-		// Assertion only works for BridgeAbstract!
-		if(in_array('BridgeAbstract', class_parents($bridgeName))) {
+			foreach ($paramsSeen as $seen) {
+				$this->assertNotEquals($seen, $params, 'same set of parameters not allowed');
+			}
+			$paramsSeen[] = $params;
 
-			$ref = new ReflectionClass($bridgeName);
-			$constants = $ref->getConstants();
+			foreach ($params as $field => $options) {
+				$this->assertInternalType('string', $field, $field . ': invalid id');
+				$this->assertNotEmpty($field, $field . ':empty id');
 
-			$ref = new ReflectionClass('BridgeAbstract');
-			$parent_constants = $ref->getConstants();
+				$this->assertInternalType('string', $options['name'], $field . ': invalid name');
+				$this->assertNotEmpty($options['name'], $field . ': empty name');
 
-			foreach($parent_constants as $key => $value) {
+				if (isset($options['type'])) {
+					$this->assertInternalType('string', $options['type'], $field . ': invalid type');
+					$this->assertContains($options['type'], $allowedTypes, $field . ': unknown type');
 
-				$this->assertArrayHasKey($key, $constants, 'Constant ' . $key . ' missing in ' . $bridgeName);
+					if ($options['type'] == 'list') {
+						$this->assertArrayHasKey('values', $options, $field . ': missing list values');
+						$this->assertInternalType('array', $options['values'], $field . ': invalid list values');
+						$this->assertNotEmpty($options['values'], $field . ': empty list values');
 
-				// Skip optional constants
-				if($key !== 'PARAMETERS' && $key !== 'CACHE_TIMEOUT') {
-					$this->assertNotEquals($value, $constants[$key], 'Constant ' . $key . ' missing in ' . $bridgeName);
+						foreach ($options['values'] as $valueName => $value) {
+							$this->assertInternalType('string', $valueName, $field . ': invalid value name');
+						}
+					}
 				}
 
-			}
-
-		}
-
-	}
+				if (isset($options['required'])) {
+					$this->assertInternalType('bool', $options['required'], $field . ': invalid required');
+				}
 
-	private function isFunctionMemberOf($bridgeName, $functionName){
+				if (isset($options['title'])) {
+					$this->assertInternalType('string', $options['title'], $field . ': invalid title');
+					$this->assertNotEmpty($options['title'], $field . ': empty title');
+				}
 
-		$bridgeReflector = new ReflectionClass($bridgeName);
-		$bridgeMethods = $bridgeReflector->GetMethods();
-		$bridgeHasMethod = false;
+				if (isset($options['pattern'])) {
+					$this->assertInternalType('string', $options['pattern'], $field . ': invalid pattern');
+					$this->assertNotEquals('', $options['pattern'], $field . ': empty pattern');
```

Comment:
This is intentional, because there are bridges that use '0' as a string value. They get considered being empty even though they're not (see commit bbf18cfdc9c87ec8192868284c167dcdce1397d9).